[sdf] Fix Sdf_Children<ChildPolicy>::Find to return correct value

nvidia-jomiller opened the pull request:
In most cases, Sdf_Children<ChildPolicy>::Find would return the incorrect value of 0 instead the size of _childNames. This would incorrectly index into the first element when a match was not found

### Description of Change(s)

### Link to proposal ([if applicable](https://openusd.org/release/contributing_to_usd.html#step-1-get-consensus-for-major-changes))

### Fixes Issue(s)

### Checklist

[x] I have created this PR based on the dev branch

[x] I have followed the [coding conventions](https://openusd.org/release/api/_page__coding__guidelines.html)

[x] I have added unit tests that exercise this functionality (Reference: 
[testing guidelines](https://openusd.org/release/api/_page__testing__guidelines.html))

[x] I have verified that all unit tests pass with the proposed changes

[x] I have submitted a signed Contributor License Agreement (Reference: 
[Contributor License Agreement instructions](https://openusd.org/release/contributing_to_usd.html#contributor-license-agreement))

nvidia-jomiller committed:
[sdf] Fix Sdf_Children<ChildPolicy>::Find to return correct value

In most cases, Sdf_Children<ChildPolicy>::Find would return the incorrect value of 0 instead the size of _childNames.
This would incorrectly index into the first element when a matching element was not found

jesschimein commented:
Filed as internal issue #USD-10583

(This is an automated message. See [here](https://openusd.org/release/contributing_to_usd.html#github-issues) for more information.)

jesschimein commented:
/AzurePipelines run

sunyab commented:
Hey @nvidia-jomiller, 

I'm not quite seeing the bug you mentioned in your PR description. Prior to your change, in the case where the given `key` isn't found, the index `i` would have been incremented to `_childNames.size()` and then returned. The test case you added worked both with and without your change. AFAICT, the only functional change is that in the failed verify case we now return `_childNames.size()` instead of 0.

Am I missing something?

nvidia-jomiller commented:
Hey @sunyab , thanks for taking a look!

The change was mostly facilitated by another task but yeah, now that I read through it line by line I see functionally it's equivalent. What I was mostly going from is checks in [SdfChildrenView](https://github.com/PixarAnimationStudios/OpenUSD/blob/dev/pxr/usd/sdf/childrenView.h#L555) and code docs in [Sdf_Children](https://github.com/PixarAnimationStudios/OpenUSD/blob/dev/pxr/usd/sdf/children.h#L65). I didn't see any tests specifically testing it so I modified and added a test. I think this might just be a readability thing where I was expecting to see `_childNames.size()` returned. But I'm happy to close this out

sunyab commented:
Ok, thanks for confirming! Please leave this open for now, if nothing else I think we can merge in the test since more tests are always appreciated.
